cg/meta/upload/nipt/nipt.py

```diff
@@ -50,18 +48,25 @@ def set_dry_run(self, dry_run: bool) -> None:
         LOG.info("Set dry run to %s", dry_run)
         self.dry_run = dry_run
 
-    def flowcell_passed_qc_value(self, case_id: str, q30_threshold: float) -> bool:
-        """Check average Q30 and of the latest flow cell related to a case."""
-        latest_flow_cell: Flowcell = self.status_db.get_latest_flow_cell_on_case(family_id=case_id)
-        flow_cell_reads_and_q30_summary: Dict[
-            str, Union[int, float]
-        ] = self.stats_api.flow_cell_reads_and_q30_summary(flow_cell_name=latest_flow_cell.name)
-        return flow_cell_reads_and_q30_summary[
-            "q30"
-        ] >= q30_threshold and flow_cell_reads_and_q30_summary[
-            "reads"
-        ] >= self.status_db.get_ready_made_library_expected_reads(
-            case_id=case_id
+    def flowcell_passed_qc_value(self, case_id: str, q30_threshold: float):
+        """
+        Check the average Q30 and the total number of reads for each sample
+        in the latest flow cell related to a case.
+        """
+        flow_cell: Flowcell = self.status_db.get_latest_flow_cell_on_case(family_id=case_id)
+        flow_cell_summary: FlowCellQ30AndReads = FlowCellQ30AndReads(
+            q30=self.status_db.get_average_passing_q30_for_samples_on_flow_cell(
+                flow_cell_name=flow_cell.name
+            ),
+            reads=self.status_db.get_number_of_reads_for_flow_cell(flow_cell_name=flow_cell.name),
+        )
+        return (
```

Comment:
```suggestion
        return (
            flow_cell_summary.q30_above_threshold(threshold=q30_threshold) and
            flow_cell_summary.reads_above_threshold(
                threshold=self.status_db.get_ready_made_library_expected_reads(case_id=case_id)
            )
```

Reply:
If both returns are true the return is true, if one is false its false

Reply:
Now that I think about it - Was there ever a log message for when one of the values failed? I really think we should log it because we would like to know if we skip the upload for some reason.

Reply:
I think this is what it was before also, right? Both need to be TRUE for the QC to pass I imagined. Can you confirm?